### DISCUSSION TITLE
While editing a model with media - adding new media does not create conversions or image from pdf

### DISCUSSION THREAD

**Discussion by trevorpan** [spatie/laravel-medialibrary#1943] [Q&A]
Has anyone had a situation where adding new media to existing model does not create conversions or take an image from a pdf?

I copied the code for adding media from the `store()` method. So it's strange when updating using `update()` does not trigger conversions. 

Is this an issue because it is a `PATCH` / `POST` request?

**Comment by trevorpan** [comment]
Sort of strange. I shut down the machine and reopened today. The conversions are all there!

What's weird though is I tried uploading again on a new model and the conversions are not created "yet". This is a rather large file 8.5mb and it has some blacked out areas (so they may be like a layer thought I've flattened them in acrobat).

Is there a flatten command? Or how would you handle this.

It appears the machine is just taking a long time to make the conversions...

**Comment by francoism90** [comment]
Could you please share your code? I don't understand what you mean, do you mean the regenerate command? You need to call that command if you want to re-run the conversion.

**Reply by trevorpan** [reply]
Hi @francoism90 

Thank you for the note! I meant if someone made a post and didn't upload an image or pdf at the time of creation, then came back to add a file to an existing model the conversions weren't created.

It was a real pain`

So, I found after finally getting a good error the imagick policy.xml file did not have this setting:
```
<policy domain="coder" rights="read | write" pattern="PDF" />
```
Basically, the pdf-to-image package was getting hung up on the rights. In addition the local env had a sync setting for the queue rather than redis and horizon. I briefly switch it to test the functionality - and it works great.
```
 /**
     * Register the conversions that should be performed.
     *
     * @param Media|null $media
     * @return void
     * @throws \Spatie\Image\Exceptions\InvalidManipulation
     */
    public function registerMediaConversions(Media $media = null)
    {
        $this->addMediaConversion('thumbnail')
            ->width(215)
            ->height(140)
            ->sharpen(10)
            ->apply()
            ->withResponsiveImages()
            ->performOnCollections('document')
            ->queued();

        $this->addMediaConversion('jobfullpage')
            ->fit(Manipulations::FIT_CROP, 320, 240)
            ->apply()
            ->withResponsiveImages()
            ->performOnCollections('document')
            ->queued();
    }
```

If you are interested here's some more details... thank you again.
https://github.com/spatie/pdf-to-image/issues/153
